### DISCUSSION TITLE
Using wherehas fails

### DISCUSSION THREAD

**Discussion by unknown1337** [CodeDredd/pinia-orm#292] [Q&A] [answered]
**Situation**
Creative:  @HasOne(() => Ad, 'creativeId') declare ad: Ad
 Ad:   @Num(null) declare extId: number

**data**
`    return useRepo(CreativeRepo).with('ad', (query) => {
      query.where('extId', 6283472664440)
    }).get()`
![image](https://user-images.githubusercontent.com/5968968/188315889-86d8f052-65dd-4223-ab38-9c21bd6b3a01.png)

1. multiple ads, with  entry that has extId == 6283472664440
2. multiple creatives (1 per ad)

**Goal**
ONLY retreive Creative where (ad) extId == 6283472664440. Desired outcome: 
![image](https://user-images.githubusercontent.com/5968968/188316152-c6ab90c3-97c1-493e-b64e-ffeda84fc631.png)

**failing attempt**
```
 return useRepo(CreativeRepo).whereHas('ad', (query) => {
      query.where('extId', 6283472664440)
    }).get()
```
error: `Uncaught (in promise) Error: [Pinia ORM] Relation ad not found in model:  creative`

**Question**
what could I be doing wrong? in section **data** ive shown that the relation exists

Thanks!! :)

**apendix: Models**
creative
```
import { Model } from "pinia-orm";
import { HasOne, Num, Str } from "pinia-orm/dist/decorators";
import Ad from "./adModel";

export default class Creative extends Model {
  static entity = 'creative'
  static primaryKey = 'extId'

  @Str('') declare extId: string
  @Str('') declare name: string
  @Str('') declare imageUrl: string
  @Str('') declare thumbnailUrl: string
  @Num(null) declare videoId: number
  @HasOne(() => Ad, 'creativeId') declare ad: Ad
}
```
ad
```
import { Model } from "pinia-orm";
import { Str, Num, BelongsTo, BelongsToMany } from "pinia-orm/dist/decorators";
import Adset from "./adsetModel";
import Campaign from "./campaignModel";
import Creative from "./creativeModel";
import Group from "./groupModel";
import Group_Ad from "./group_adModel";

export default class Ad extends Model {
  static entity = 'ad'
  static primaryKey = 'extId'

  @Str('') declare extId: string
  @Str('') declare campaignId: string
  @Str('') declare adsetId: string
  @Str('') declare creativeId: string
  @Str('') declare name: string
  @Str('') declare platform: string
  @BelongsTo(() => Adset, 'adsetId') declare adset: Adset
  @BelongsTo(() => Campaign, 'campaignId') declare campaign: Campaign
  @BelongsTo(() => Creative, 'creativeId') declare creative: Creative
  @BelongsToMany(() => Group, () => Group_Ad, 'adExtId', 'groupId') declare groups: Group[] | undefined


}
```

**Comment by CodeDredd** [comment, marked as answer]
uff 10 days ago ^^....i am soon on it. Too much work last days 😉

**Reply by CodeDredd** [reply]
ok i think i know what the reason is...its a small bug....fixing it for 1.0.0

**Reply by CodeDredd** [reply]
fixed with #342 . Thanks for finding the bug

**Reply by unknown1337** [reply]
Thanks a lot for the bugfix!
